engine/src/main/java/org/hibernate/validator/internal/engine/cascading/LegacyIterableValueExtractor.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.internal.engine.cascading;
+
+import org.hibernate.validator.spi.cascading.ValueExtractor;
+
+class LegacyIterableValueExtractor implements ValueExtractor<Iterable<?>> {
+
+	static final LegacyIterableValueExtractor INSTANCE = new LegacyIterableValueExtractor();
+
+	private LegacyIterableValueExtractor() {
+	}
+
+	@Override
+	public void extractValues(Iterable<?> originalValue, ValueReceiver receiver) {
+		receiver.value( originalValue, null );
```

Comment:
The legacy one will be used for `@Valid Set<Order> orders`, and the new one will be used for `Set<@Valid Order> orders`. The difference being that the former would validate any property constraints on a custom `Set` implementation, whereas the latter would only validate the actual iterable elements.

Selection is happening via the type parameter to which extractors are bound: the non-legacy one is bound to `<T>` of Iterable, the legacy one is bound to an internal pseudo type parameter `ANNOTATED_ELEMENT`. Depending on where `@Valid` is spotted, one or the other will be used.